src/sources/aws_s3/sqs.rs

```diff
@@ -49,21 +50,27 @@ static SUPPORTED_S3_EVENT_VERSION: Lazy<semver::VersionReq> =
 //
 // TODO: It seems awfully likely that we could re-use the existing configuration type for the `aws_sqs` source in some
 // way, given the near 100% overlap in configurable values.
+#[serde_as]
 #[configurable_component]
 #[derive(Clone, Debug, Derivative)]
 #[derivative(Default)]
 #[serde(deny_unknown_fields)]
 pub(super) struct Config {
     /// The URL of the SQS queue to poll for bucket notifications.
+    #[configurable(metadata(
+        docs::examples = "https://sqs.us-east-2.amazonaws.com/123456789012/MyQueue"
+    ))]
     pub(super) queue_url: String,
 
     /// How long to wait while polling the queue for new messages, in seconds.
     ///
-    /// Generally should not be changed unless instructed to do so, as if messages are available, they will always be
-    /// consumed, regardless of the value of `poll_secs`.
+    /// Generally should not be changed unless instructed to do so, as if messages are available,
+    /// they will always be consumed, regardless of the value of `poll_secs`.
     // NOTE: We restrict this to u32 for safe conversion to i64 later.
+    // TODO: Duration?
```

Comment:
Happy to take opinions/advice here, but the aws sdk takes a signed integer which made the `Duration`-izing awkward here.

Reply:
IMO no need to make it Duration if we aren't using it as a Duration downstream.
I guess could mention in a non-doc comment this reason.

Reply:
Addressed in [7fa8b61](https://github.com/vectordotdev/vector/pull/15836/commits/7fa8b61cfcbbb178bec95810849aed048ea47ba8)